deny.toml

```diff
@@ -10,15 +10,15 @@ targets = [
 vulnerability = "deny"
 unmaintained = "deny"
 yanked = "deny"
-ignore = [
-]
+ignore = []
 
 # https://embarkstudios.github.io/cargo-deny/checks/licenses/cfg.html
 [licenses]
 allow = [
     "Apache-2.0 WITH LLVM-exception",
     "Apache-2.0",
     "MIT",
+    "Unicode-DFS-2016",
```

Comment:
Could we add a comment why this is needed?

Probably fine from first principles but any addition here does mean we need to update downstream users' allowlists as well (e.g. Cranelift) which is sometimes a little tricky / requires consultation to be sure.

If my guess that this comes from `pretty_env_logger` is correct, perhaps could we use just `env_logger`?

Reply:
It actually comes from `unicode-ident`, which comes from `syn`, which is itself used by `serde-derive`. I think this appears here because regalloc2-tool enables the `enable-serde` feature of the crate. It shouldn't affect any downstream users if they don't have `enable-serde` enabled.